core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -105,22 +112,26 @@ func (c *OutgoingConnectorHandler) HandleSingleNodeRequest(ctx context.Context,
 
 	select {
 	case resp := <-ch:
-		l.Debugw("received response from gateway", "gatewayID", selectedGateway)
+		lggr.Debugw("received response from gateway")
 		return resp, nil
 	case <-ctx.Done():
 		return nil, ctx.Err()
 	}
 }
 
+// HandleGatewayMessage processes incoming messages from the Gateway,
+// which are in response to a HandleSingleNodeRequest call.
 func (c *OutgoingConnectorHandler) HandleGatewayMessage(ctx context.Context, gatewayID string, msg *api.Message) {
 	body := &msg.Body
 	l := logger.With(c.lggr, "gatewayID", gatewayID, "method", body.Method, "messageID", msg.Body.MessageId)
+
 	if !c.rateLimiter.Allow(body.Sender) {
 		// error is logged here instead of warning because if a message from gateway is rate-limited,
 		// the workflow will eventually fail with timeout as there are no retries in place yet
-		c.lggr.Errorw("request rate-limited")
+		l.Errorw("request rate-limited")
 		return
```

Comment:
Technically out of scope, but could we return an error here back to the caller? I think this results in nicer UX by telling users they were rate limited rather than returning a timeout.

Reply:
Agreed. Changing.

Reply:
Changed here https://github.com/smartcontractkit/chainlink/pull/16672/commits